Sources/SwiftGRPCNIO/GRPCHTTPProtocolSwitcher.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import NIO
+import NIOHTTP1
+import NIOHTTP2
+
+/// Channel handler that creates different processing pipelines depending on whether
+/// the incoming request is HTTP 1 or 2.
+class GRPCHTTPProtocolSwitcher {
+  private let handlersInitializer: ((Channel) -> EventLoopFuture<Void>)
+
+  public init(handlersInitializer: (@escaping (Channel) -> EventLoopFuture<Void>)) {
```

Comment:
Would you mind making the entire class `public` or `open` as well, so low-level users of this library can make use of it as well?

Reply:
done